src/parser/expression.js

```diff
@@ -301,6 +301,44 @@ export default class ExpressionParser extends LValParser {
         node.object = base;
         node.callee = this.parseNoCallExpr();
         return this.parseSubscripts(this.finishNode(node, "BindExpression"), startPos, startLoc, noCalls);
+
+      } else if (this.match(tt.questionDot)) {
+        if (!this.hasPlugin("optionalChaining")) {
+          this.raise(startPos, "You can only use optional-chaining when the 'optionalChaining' plugin is enabled.");
+        }
+
+        if (noCalls && this.lookahead().type == tt.parenL) {
+          return base;
+        }
+        this.next();
+
+        const node = this.startNodeAt(startPos, startLoc);
+
```

Comment:
I was trying to keep the property ordering consistent between optional and non-optional branches. See the `#parseNew` as well.